### PR TITLE
Improve checkbinary to attempt to find heroic binary on PATH

### PR DESCRIPTION
**Pull request by afwolfe**
Hello, thanks for this great tool. I've been using it for awhile now on my Fedora Linux system to quickly launch games from Steam while using the RPM-packaged version of Heroic Games Launcher from the recommended COPR repo ([atim/heroic-games-launcher](https://copr.fedorainfracloud.org/coprs/atim/heroic-games-launcher/)).

Unfortunately, due to the hard-coding of the expected paths of Heroic in HeroicBashLauncher, it wasn't working on my system at first because the RPM version of Heroic is installed to `/usr/lib64/heroic-games-launcher-bin/`. 

This PR attempts to fix this issue and allow for more "diverse" Heroic installations in the future by:
* Attempting to find `heroic` on the PATH before going back to the old defaulting logic.
* Including unit tests for checkbinary.py to confirm that each of the correct paths are returned in a few different scenarios.
    * These can be invoked with `python -m unittest tests/*.py`
    * A side-effect of making the modules testable required reorganizing the main `HeroicBashLauncher.py` to nest its code under an `if __name__ == "__main__":` block and converting a few of the checks in `settings.py` to be functions that are now called from this main method.

### CODE REVIEW

**Comment by redromnon** [comment]
Thanks a lot for this @afwolfe!

File: func/checkbinary.py [hunk 1]
```diff
@@ -19,38 +21,43 @@ def getbinary(gametype):
...
+        elif os.path.exists("/opt/Heroic"): # Default to /opt/Heroic
+            heroic_base_path = "/opt/Heroic"
+        elif configpath.is_flatpak or os.path.exists("/app/bin/heroic"): #System or Flatpak-env path
+            heroic_base_path = "/app/bin/heroic"
+        
+        if heroic_base_path:
+
+            heroic_resources_path = os.path.join(heroic_base_path, resources_bin_path)
+
+            if os.path.exists(heroic_resources_path):
```
Comment (redromnon) [inline func/checkbinary.py:48]: There's an issue here in case of Heroic Flatpak. 

The path check on line 48 will fail as `/app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/<executable>` is only present in the Flatpak environment.

So `binary` will remain as an empty string and the executable path won't be displayed in the bash launch script.

Comment (redromnon) [reply func/checkbinary.py:48]: I just tested this on my VM where I've got Heroic Flatpak installed and the script fails to launch any game as the binary path is empty. As a workaround, I commented this check and it worked perfectly fine.

Maybe you can keep this check for the non-Flatpak versions of Heroic?

Comment (afwolfe) [reply func/checkbinary.py]: Thanks for the clarification! I wasn't as familiar with this path but that makes sense. I am going to remove the extra check since the other branches actually already check `os.path.exists()` before continuing. I also updated the unit test to reflect this. 

Talking about this made me think of a question: should `configpath.is_flatpak` be checked first in this method? While unlikely, someone could have Heroic installed natively and as a Flatpak, and the method here would return the system installation of Heroic, but the paths being returned for the paths in configpath will always be for the Flatpak. Depending on the expected behavior, I could add a unit test to check this edge case too.

Also, would it be okay to update build.yml as part of this PR to run unit tests automatically?

Comment (redromnon) [reply func/checkbinary.py]: Good point. I guess you reorder the checking accordingly.

Comment (afwolfe) [reply func/checkbinary.py]: @redromnon I pushed a change yesterday to prefer the Flatpak, but realized handling that edge case by making the Flatpak check first could also cause other users to be impacted by #102.

I have pushed another change to attempt to fix #102 by detecting when the Heroic Flatpak is installed.
Let me know if you'd prefer I save that for another PR and I can undo it and 072f576 for now to keep the behavior closer to what's in main today.

Comment (redromnon) [reply func/checkbinary.py]: @afwolfe I guess it'll be better to fix that issue in another PR. I had completely forgotten about that issue.

**Comment by afwolfe** [comment]
Hello @redromnon, I reverted the changes we discussed for now and this is ready for a (hopefully) final review. Please take your time, and thanks again for your help and feedback.

I'll open another PR for the Flatpak detection fix once this is merged so I can build off of these changes.

**Review by redromnon — APPROVED** [review]
LGTM :+1:
